src/app/worker/suggestion_processor.py

```
@@ -0,0 +1,323 @@
+            # 3. Save each translated verse back via API. Guard each item
+            # individually — one hallucinated/malformed verse_id from the LLM
+            # should not fail the whole batch (see review finding #4).
+            items = []
+            parsed_verse_numbers = set()
+            for item in result.translations:
+                try:
+                    verse_num = int(item.verse_id.split("_")[-1])
+                except (ValueError, AttributeError):
+                    logger.warning(
+                        f"Job {job.id}: skipping unparseable verse_id "
+                        f"{item.verse_id!r} from LLM response."
```

Ruff (0.15.20)

[warning] Logging statement uses f-string

(G004)